packages/core/src/transports/multiplexed.ts

```diff
@@ -0,0 +1,84 @@
+import type {
+  BaseTransportOptions,
+  Envelope,
+  Event,
+  EventItem,
+  Transport,
+  TransportMakeRequestResponse,
+} from '@sentry/types';
+import { dsnFromString, forEachEnvelopeItem } from '@sentry/utils';
+
+import { getEnvelopeEndpointWithUrlEncodedAuth } from '../api';
+
+interface MatchParam {
+  // The envelope to be sent
+  envelope: Envelope;
+  // A function that returns an event from the envelope if one exists
+  getEvent(): Event | undefined;
+}
+
+type Matcher = (param: MatchParam) => string[];
+
+function eventFromEnvelope(env: Envelope): Event | undefined {
+  let event: Event | undefined;
+
+  forEachEnvelopeItem(env, (item, type) => {
+    if (type === 'event' || type === 'transaction') {
+      event = Array.isArray(item) ? (item as EventItem)[1] : undefined;
+    }
+    // bail out if we found an event
+    return !!event;
+  });
+
+  return event;
+}
+
+/**
+ * Creates a transport that can send events to different DSNs depending on the envelope contents.
+ */
+export function makeMultiplexedTransport<TO extends BaseTransportOptions>(
+  createTransport: (options: TO) => Transport,
+  matcher: Matcher,
+): (options: TO) => Transport {
+  return options => {
+    const fallbackTransport = createTransport(options);
+    const otherTransports: Record<string, Transport> = {};
+
+    function getTransport(dsn: string): Transport {
+      if (!otherTransports[dsn]) {
+        const url = getEnvelopeEndpointWithUrlEncodedAuth(dsnFromString(dsn));
+        otherTransports[dsn] = createTransport({ ...options, url });
+      }
+
+      return otherTransports[dsn];
+    }
+
+    async function send(envelope: Envelope): Promise<void | TransportMakeRequestResponse> {
+      function getEvent(): Event | undefined {
+        return eventFromEnvelope(envelope);
+      }
+
+      const transports = matcher({ envelope, getEvent }).map(([dsn]) => getTransport(dsn));
+
+      // If we have no transports to send to, use the fallback transport
+      if (transports.length === 0) {
+        transports.push(fallbackTransport);
+      }
+
+      const results = await Promise.all(transports.map(transport => transport.send(envelope)));
+
+      return results[0];
```

Comment:
Not sure of a better solution than this 🤔

Reply:
I think this is fine!